### ISSUE TITLE
Using K-FAC training BERT-Large model

### ISSUE THREAD

**Issue by RuidongYan666** [gpauloski/kfac-pytorch#117]
Hi, gpauloski. Thanks for helping.

Recently, we have done some experiments, and the following questions need to be solved.

1.  (**Without K-FAC,  e.g., using LAMB in phase 1 and phase 2**): In phase 1, we train the Bert-large model 7038 steps and the final average_loss is 1.6962 (FP16). Training time is 69.3353 hours.   In phase 2, we train the model 1563 steps and the final average_loss is 1.4426 (FP16). Training time is 36.2413 hours.   We then fine-tune the pretrained BERT model for 3 epochs using
the SQuAD dataset. The F1-score is 89.55.  

The training times and F1-score  are somewhat different from those reported in your paper(SC'21, Table 3). For example, 
training time in your paper in phase 2 with LAMB is 47.7 hours, however, we report that time is 36.2413 hours. The F1-score
in your paper is 90.8, however, our F1-score is noly 89.55.  Note that we run experiments on 8 A100 GPUs.
So,  how can we improve F1-scores (e.g., 90.8)?  Whether there are any special skills in the downstream task fine-tuning or pre-training? 

2. (**With K-FAC,  e.g.  using LAMB in phase 1 and using K-FAC in phase 2**):  Keep the same 7038.pt in phase 1. 
When we train the Bert-large model with 800/1000/1200 steps  in phase 2, the F1-score are 89.02/88.27/89.32, respectively.  
We can see the F1-scores of K-FAC are less than those of LAMB (i.e., 89.55). What should we do to improve the F1-score of K-FAC over LAMB?  By the way, the training time of K-FAC with 800/1000/1200 steps in phase 2 is less than LAMB.

Best regards,
Ruidong

**Comment by gpauloski** [comment]
Hi Ruidong, could you share the parameters you used in each case? I can give a better answer with more details about your training environment, hyper parameters, data sets, K-FAC parameters, etc.

**Comment by RuidongYan666** [comment]
Hi, gpauloski.   Please tell me your email address  so that I can send you some log files of our experiments.

**Comment by gpauloski** [comment]
jgpauloski@uchicago.edu

**Comment by gpauloski** [comment]
Hi, @RuidongYan666! I reviewed the logs you emailed me, but I'm going to keep the conversation here in case anyone else runs into a similar issue.

Your version of the English Wikipedia and Toronto BookCorpus dataset appears to be much smaller than ours. Your logs list `Samples in dataset: 69004805`, whereas ours is `Samples in dataset: 179962902`. That's less than half the size of ours and would definitely account for the discrepancies in the final loss of your phase 1 training. Otherwise it looks like all of your phase 1 parameters match ours.

I'm going to close this issue as I'm pretty confident that the dataset size discrepancy is the problem, and per https://github.com/gpauloski/llm-pytorch/issues/56#issuecomment-1641285170 it seems the issues are arising from the phase 1 training and not the phase 2 training with K-FAC.

If you have any other questions, feel free to ask.
